pkgs/applications/science/robotics/mujoco/default.nix

```diff
@@ -1,122 +1,230 @@
+      ln -s ${pin.lodepng} build/_deps/lodepng-src
+      ln -s ${pin.qhull} build/_deps/qhull-src
+      ln -s ${pin.tinyobjloader} build/_deps/tinyobjloader-src
+      ln -s ${pin.tinyxml2} build/_deps/tinyxml2-src
+      ln -s ${pin.marchingcubecpp} build/_deps/marchingcubecpp-src
+    '';
+
+    meta = with lib; {
+      description =
+        "Multi-Joint dynamics with Contact. A general purpose physics simulator.";
+      homepage = "https://mujoco.org/";
+      license = licenses.asl20;
+      maintainers = with maintainers; [ samuela tmplt ];
+    };
   };
 
-  patches = [ ./dependencies.patch ];
-
-  nativeBuildInputs = [ cmake git ];
-
-  buildInputs = [
-    libGL
-    xorg.libX11
-    xorg.libXcursor
-    xorg.libXext
-    xorg.libXi
-    xorg.libXinerama
-    xorg.libXrandr
-  ];
-
-  # Move things into place so that cmake doesn't try downloading dependencies.
-  preConfigure = ''
-    mkdir -p build/_deps
-    ln -s ${abseil-cpp} build/_deps/abseil-cpp-src
-    ln -s ${benchmark} build/_deps/benchmark-src
-    ln -s ${ccd} build/_deps/ccd-src
-    ln -s ${eigen3} build/_deps/eigen3-src
-    ln -s ${glfw3} build/_deps/glfw3-src
-    ln -s ${googletest} build/_deps/googletest-src
-    ln -s ${lodepng} build/_deps/lodepng-src
-    ln -s ${qhull} build/_deps/qhull-src
-    ln -s ${tinyobjloader} build/_deps/tinyobjloader-src
-    ln -s ${tinyxml2} build/_deps/tinyxml2-src
-  '';
-
-  meta = with lib; {
-    description = "Multi-Joint dynamics with Contact. A general purpose physics simulator.";
-    homepage = "https://mujoco.org/";
-    license = licenses.asl20;
-    maintainers = with maintainers; [ samuela ];
+  python-bindings = buildPythonPackage rec {
+    inherit pname version;
+    format = "pyproject";
+
+    src = fetchPypi {
+      inherit pname version;
+      hash = "sha256-2ImKDfm9hCORaH7iou/Lp979VSCugCM/Gqnb/CkTeOw=";
+    };
+
+    nativeBuildInputs = [ cmake setuptools setuptools-scm wheel ];
+    dontUseCmakeConfigure = true;
+    buildInputs = [
+      mujoco
+
+      # non-numerical
+      pybind11
+      glfw-lib
+      glfw
+    ];
+    propagatedBuildInputs = [ glfw ];
+
+    env.MUJOCO_PATH = "${mujoco}";
+    env.MUJOCO_PLUGIN_PATH = "${mujoco}/lib";
+
+    preConfigure =
+      # Use system packages for pybind, glfw
+      ''
+        sed -i '185s/.*/  ON/' mujoco/CMakeLists.txt # pybind
+        sed -i '79s/.*/  ON/'  mujoco/simulate/cmake/SimulateDependencies.cmake # glfw
+      '' +
+      # Use non-system eigen3, lodepng, abseil: Remove mirror info and prefill
+      # dependency directory. $build from setuptools.
+      (let
+        # E.g. 3.11.2 -> "311"
+        pythonVersionMajorMinor = with lib.strings;
+          with lib.lists;
+          concatStrings (reverseList
+            (drop 1 (reverseList (splitString "." python3.version))));
+      in ''
+        sed -i '140d;141d;142d;143d;173d;174d;175d;176d' mujoco/CMakeLists.txt # abseil, eigen
```

Comment:
This removes
```
  GIT_REPO
  https://github.com/abseil/abseil-cpp
  GIT_TAG
  fb3621f4f897824c0dbe0615fa94543df6192f30 # LTS 20230802.1
```
and similar for abseil and eigen.

There is probably a cleaner call.